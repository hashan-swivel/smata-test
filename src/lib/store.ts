// In your 'store.ts' or where you set up your Redux store
import { modalReducer } from '@/lib/features/modal/modalSlice';
import { combineReducers, configureStore } from '@reduxjs/toolkit';

const rootReducer = combineReducers({
  modal: modalReducer
});

export const store = configureStore({
  reducer: rootReducer
});

export type AppStore = typeof store;
export type AppState = ReturnType<AppStore['getState']>;
export type AppDispatch = AppStore['dispatch'];
