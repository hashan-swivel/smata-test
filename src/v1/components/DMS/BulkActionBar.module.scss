@import '@/sass/base/_typography.module.scss';
@import '@/sass/base/_mixins.module.scss';
@import '@/sass/base/__media.module.scss';
@import '@/sass/base/__variables.module.scss';
.bulk-actions-bar {
  width: 100%;
  padding: 15px 20px;
  background: #ffffff;
  border: 1px solid #ffffff;
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.bulk-actions-bar--left,.bulk-actions-bar--right {
  display: flex;
  align-items: center;
}

.bulk-actions-bar__clear-button {
  background: none;
  border: none;
  margin-right: 30px;
  padding: 0;
  display: flex;
}

.bulk-actions-bar--right > .dropdown-wrapper > .label {
  border-color: $border;
}

.bulk-actions-button {
  font-size: inherit;
  display: flex;
  align-items: center;
  font-weight: normal;
  padding: 5px;
  text-decoration: none;
  border-radius: 4px;
  background: $white;
  color: $black;
  border: 1px solid $border;
}
