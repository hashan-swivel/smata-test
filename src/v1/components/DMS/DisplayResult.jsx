import React from 'react';
import { useDispatch } from 'react-redux';
import Router from 'next/router';
import moment from 'moment';
import './DisplayResult.module.scss';
import { setFilters } from '../../../actions/dms';

export const DisplayResult = (props) => {
  const dispatch = useDispatch();
  const { title, data, setSearchVal, searchFilters = [] } = props;

  const handleTagsMouseDown = (type, item) => {
    const filterPreviouslySet = searchFilters.some((filter) => filter.item === item);
    if (!filterPreviouslySet) {
      const clonedSearchFilters = [...searchFilters];
      clonedSearchFilters.push({ type, item });
      dispatch(setFilters(clonedSearchFilters));
    }
    setSearchVal('');
  };

  if (title === 'documents' || title === 'invoices') {
    const isDocuments = title === 'documents';

    return (
      <ul className='result-list'>
        {data.map((item) => {
          const { id, filename, category, created_at: createdAt, added_date: addedDate } = item;
          const hrefPath = category === 'invoice' ? '/invoice' : '/document-preview';
          return (
            <div
              role='presentation'
              className='document-results'
              key={id}
              onMouseDown={() => Router.push(`${hrefPath}?id=${id}`)}
            >
              <li className='document-name'>{filename}</li>
              <span className='document-number-date'>
                <li className='document-date'>
                  {moment.unix(isDocuments ? addedDate : createdAt).format('DD/MM/YYYY')}
                </li>
              </span>
            </div>
          );
        })}
      </ul>
    );
  }

  if (title === 'sp_number') {
    return (
      <div>
        {data.map((item, index) => (
          <span
            role='presentation'
            className='display-result result-container'
            key={index}
            onMouseDown={() => {
              handleTagsMouseDown(title, item);
            }}
          >
            <span className='result-name'>{item}</span>
            <span className='result-plus'>+</span>
          </span>
        ))}
      </div>
    );
  }

  if (title === 'images') {
    return (
      <div className='result-list'>
        {data.map((item) => {
          const { id, display_name: displayName, url, category, created_at: createdAt } = item;
          const hrefPath = category === 'invoice' ? '/invoice' : '/document-preview';

          return (
            <div
              role='presentation'
              className='image-search-results'
              key={url}
              onMouseDown={() => Router.push(`${hrefPath}?id=${id}`)}
            >
              <img className='image-search-display' src={url} alt={displayName} />
              <span className='image-search-list'>
                <span className='image-search-item'>
                  <span className='image-search-name'>{displayName}</span>
                </span>
                <span className='image-search-date'>
                  {moment.unix(createdAt).format('DD/MM/YYYY')}
                </span>
              </span>
            </div>
          );
        })}
      </div>
    );
  }

  if (data && title !== 'documents') {
    return (
      <div>
        {data.map((item) => (
          <span
            role='presentation'
            className='display-result result-container'
            key={item}
            onMouseDown={() => {
              handleTagsMouseDown(title, item);
            }}
          >
            {title === 'Manager' ? <span>IMAGE</span> : null}
            <span className='result-name'>{item}</span>
            <span className='result-plus'>+</span>
          </span>
        ))}
      </div>
    );
  }
  return null;
};
