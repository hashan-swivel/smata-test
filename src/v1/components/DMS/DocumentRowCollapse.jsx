import React, { useEffect, useRef, useState } from 'react';
import { Link } from '../index';

import './DocumentRowCollapse.module.scss';

const DocumentRowCollapse = ({ notes, tags, expandedRows, windowWidth }) => {
  const [isNoteTruncated, isNoteTrunctated] = useState(true);
  const [canTruncateNote, setCanTruncateNote] = useState(false);

  const noteElement = useRef();

  useEffect(() => {
    calculateCanTruncate();
  }, [noteElement, windowWidth]);

  const calculateCanTruncate = () => {
    if (noteElement.current) {
      setCanTruncateNote(noteElement.current.offsetHeight < noteElement.current.scrollHeight);
    }
  };

  return (
    <div className='document-collapse-row'>
      <div className='row-item-first-empty-col' />
      <div className={`accordion-item ${expandedRows ? '' : 'collapsed'}`}>
        <hr />

        {tags && (
          <div className='row tag-row'>
            <span className='row-item-first-col'>Tags: </span>
            <div className='row-item-second-col '>
              {tags.map((tag) => (
                <span key={tag.id} className='tag off-white'>
                  {tag.name}
                </span>
              ))}
            </div>
          </div>
        )}

        {notes && (
          <div className='row'>
            <span className='row-item-first-col'>Notes: </span>
            <div className='row-item-second-col note-wrapper'>
              <span ref={noteElement} className={isNoteTruncated ? `note note-truncated` : 'note'}>
                {notes}
                {!isNoteTruncated && canTruncateNote && (
                  <Link
                    classNameProp='show-less-link'
                    onClick={() => isNoteTrunctated(true)}
                    role='button'
                  >
                    Show less
                  </Link>
                )}
              </span>

              {isNoteTruncated && canTruncateNote && (
                <span className='show-more-link'>
                  {isNoteTruncated ? '... ' : null}
                  <Link onClick={() => isNoteTrunctated(false)} role='button'>
                    Show more
                  </Link>
                </span>
              )}
            </div>
          </div>
        )}
      </div>
    </div>
  );
};

export default DocumentRowCollapse;
