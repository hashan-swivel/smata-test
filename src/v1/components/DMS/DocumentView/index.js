export { Approvers } from './Approvers';
export { Category } from './Category';
export { FilePreview } from './FilePreview';
export { FilePreviewFooter } from './FilePreviewFooter';
export { Footer } from './Footer';
export { Header } from './Header';
export { History } from './History';
export { Tags } from './Tags';
export { Visibility } from './Visibility';
export { SharedWith } from './SharedWith';
