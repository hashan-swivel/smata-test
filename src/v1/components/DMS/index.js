export { GridView } from './GridView';
export { GridViewItem } from './GridViewItem';
export { Header } from './Header';
export { ListView } from './ListView';
export { ListViewItem } from './ListViewItem';
export { NotFound } from './NotFound';
export { Search } from './Search';
export { SearchDropdown } from './SearchDropdown';
export { SortOptions } from './SortOptions';
