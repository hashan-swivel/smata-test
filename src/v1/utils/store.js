import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import counter from '../reducers';

const store = (initialState) =>
  createStore(counter, initialState, composeWithDevTools(applyMiddleware(thunk)));

export default store;
