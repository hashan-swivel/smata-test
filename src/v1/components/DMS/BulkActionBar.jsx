import React from 'react';
import { connect } from 'react-redux';
import { bulkDeleteDocuments } from '../../../actions/dms';
import { documentActions, modalActions } from '../../../actions';
import { userOptionObj } from '../../../utils';
import { MyDropdown } from '../MyDropdown';

import './BulkActionBar.module.scss';

const selectStyles = {
  control: (provided) => ({
    ...provided,
    display: 'none'
  }),
  option: (styles, { data }) => {
    return {
      ...styles,
      color:
        data?.value === 'delete_documents' || data?.value === 'delete_invoices'
          ? '#FF0000'
          : '#353535'
    };
  },
  menu: () => ({
    boxShadow: 'inset 0 1px 0 rgba(0, 0, 0, 0.1)',
    width: 'max-content',
    minWidth: '100%'
  }),
  menuList: () => ({
    minHeight: 'fit-content'
  })
};

const BulkActionBar = ({
  documents,
  dispatch,
  selectedItemIds,
  setSelectedItemIds,
  setAllSelected,
  setReload,
  currentUser,
  viewType
}) => {
  if (selectedItemIds?.length < 1 || currentUser?.isSupport) return null;
  const isInvoice = viewType === 'invoices';
  const selectedItems = documents.filter((item) => selectedItemIds.includes(item.id));

  const deleteBulk = () => {
    dispatch(bulkDeleteDocuments(selectedItems, setReload));
  };

  const exportBulk = () => {
    dispatch(documentActions.downloadAndZipDocuments(selectedItems));
  };

  const updateBulk = () => {
    dispatch(modalActions.showModal('BULK_UPDATE_CATEGORY', { selectedItems, setReload }));
  };

  const renameBulk = () => {
    dispatch(modalActions.showModal('BULK_APPLY_SUGGESTED_NAME', { selectedItems, setReload }));
  };

  const bulkApproveInvoices = () => {
    dispatch(modalActions.showModal('BULK_APPROVE_INVOICE', { selectedItems, setReload }));
  };

  const bulkChangeInvoicePriority = (newPriority) => {
    dispatch(
      modalActions.showModal('BULK_UPDATE_PRIORITY', { selectedItems, setReload, newPriority })
    );
  };

  const bulkDropdownOptions = () => {
    const canDeleteDocument =
      currentUser?.isSystemManager ||
      (currentUser?.organisation_id && currentUser?.document_permissions?.['document.delete']);
    const canDeleteInvoice =
      currentUser?.isSystemManager ||
      (currentUser?.organisation_id && currentUser?.document_permissions?.['invoice.delete']);
    const canUpdateDocument =
      currentUser?.isOrganisationAdmin ||
      (currentUser?.organisation_id && currentUser?.document_permissions?.['invoice.override']);
    const canBulkExport = !currentUser?.isGuestUser;

    const options = canBulkExport
      ? [
          {
            label: `Export ${selectedItemIds.length} ${viewType}`,
            value: 'export',
            onClick: exportBulk
          }
        ]
      : [];

    if (!isInvoice && canUpdateDocument) {
      options.push({
        label: `Change category of ${selectedItemIds.length} documents`,
        value: 'change_documents_category',
        onClick: updateBulk
      });
      options.push({
        label: `Change name of ${selectedItemIds.length} documents`,
        value: 'change_names',
        onClick: renameBulk
      });
    }

    if (isInvoice) {
      options.push({
        label: `Approve ${selectedItemIds.length} ${viewType}`,
        value: 'approve_invoices',
        onClick: bulkApproveInvoices
      });
    }

    if (isInvoice) {
      options.push({
        label: `Remind Action Users of ${selectedItemIds.length} documents`,
        value: 'remind_invoice_action_users',
        onClick: showBulkRemindActionUsersModal
      });
      options.push({
        label: `Add priority to ${selectedItemIds.length} invoices`,
        value: 'add_priority',
        onClick: () => bulkChangeInvoicePriority(true)
      });
      options.push({
        label: `Remove priority from ${selectedItemIds.length} invoices`,
        value: 'remove_priority',
        onClick: () => bulkChangeInvoicePriority(false)
      });
    }

    if (isInvoice && canDeleteInvoice) {
      options.push({
        label: `Delete ${selectedItemIds.length} invoices`,
        value: 'delete_invoices',
        onClick: deleteBulk
      });
    }

    if (!isInvoice && canDeleteDocument) {
      options.push({
        label: `Delete ${selectedItemIds.length} documents`,
        value: 'delete_documents',
        onClick: deleteBulk
      });
    }

    return options;
  };

  const showBulkRemindActionUsersModal = async () => {
    const currentlyWithUser = selectedItems
      ?.map((item) => item.currently_with_user_details)
      ?.flat();
    dispatch(
      modalActions.showModal('REMIND_ACTION_USER', {
        bulkRemind: true,
        currentlyWithUser: currentlyWithUser?.map((user) => userOptionObj(user)),
        selectedItems
      })
    );
  };

  return (
    <div className='bulk-actions-bar'>
      <div className='bulk-actions-bar--left'>
        <button
          type='button'
          className='icon icon-cross-red bulk-actions-bar__clear-button'
          onClick={() => {
            setAllSelected(false);
            setSelectedItemIds([]);
          }}
        />
        <strong>{selectedItemIds.length} items selected</strong>
      </div>

      <div className='bulk-actions-bar--right'>
        <MyDropdown
          dropdownProps={{
            dropdownClassName: 'bulk-actions-button',
            dropdownText: 'Bulk Actions',
            showChevronDown: true
          }}
          selectProps={{ styles: selectStyles, options: bulkDropdownOptions() }}
        />
      </div>
    </div>
  );
};

export default connect((state) => state.dms)(BulkActionBar);
