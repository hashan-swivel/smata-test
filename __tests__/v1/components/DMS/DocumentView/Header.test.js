import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { Header } from '../../../../src/components/v1/DMS/DocumentView';
import renderer from 'react-test-renderer';
import store from '../../../../src/utils/store';
import {
  authMock,
  buildingProfileMock,
  documentDetailsMock,
  invoiceHeaderProps,
  invoiceMock,
  initialStateMock,
  sitePlansMock
} from '../../../../__mocks__';

const initialState = {
  ...initialStateMock,
  auth: authMock,
  buildingProfile: {
    building: buildingProfileMock
  },
  dms: {
    currentDocument: invoiceMock
  },
  spNumbers: {
    orgSpNumbers: sitePlansMock
  }
};

const props = {
  ...invoiceHeaderProps,
  ...documentDetailsMock,
  status: 'under_review',
  lot: documentDetailsMock.lot,
  name: documentDetailsMock.name,
  toggleEditMode: jest.fn(),
  invoice: invoiceMock.invoice,
  buildingData: buildingProfileMock,
  buildingProfile: buildingProfileMock,
  ownerId: invoiceMock.owner_id,
  strataManager: buildingProfileMock.strata_manager,
  currentUser: authMock.currentUser,
  spList: sitePlansMock,
  sharedWith: documentDetailsMock.sharedWith,
  openModal: jest.fn(),
  setModalType: jest.fn(),
  setKeepDropdownOpen: jest.fn(),
  isInvoiceOverrider: true,
  isUnchangeableSourceType: false
};

const mountComponent = (props, state = initialState) => {
  const mockStore = store(state);

  return mount(
    <Provider store={mockStore}>
      <Header {...props} />
    </Provider>
  );
};

describe('InvoicePreview > Header', () => {
  describe('Approve invoice button', () => {
    const target = 'button.approve-invoice-button';

    it('will not render', () => {
      const subject = mountComponent({
        ...props,
        hasVoted: true,
        canVote: true,
        isWithCurrentUser: false
      });
      expect(subject.find(target).length).toEqual(0);
    });

    it('will render', () => {
      const subject = mountComponent({ ...props });

      expect(subject.find(target).length).toEqual(1);
    });

    it('can click', () => {
      const subject = mountComponent({ ...props });

      subject.find(target).simulate('click');
      expect(props.openModal).toHaveBeenCalledTimes(1);
      expect(props.setModalType).toHaveBeenCalledWith('approve');
    });
  });

  describe('override approve button', () => {
    const target = 'button.override-approve-button';

    it('will not render', () => {
      const subject = mountComponent({ ...props });
      expect(subject.find(target).length).toEqual(0);
    });

    it('will render', () => {
      const subject = mountComponent({
        ...props,
        hasVoted: true,
        canVote: true,
        isWithCurrentUser: false
      });

      expect(subject.find(target).length).toEqual(1);
    });

    it('can click', () => {
      const subject = mountComponent({
        ...props,
        hasVoted: true,
        canVote: true,
        isWithCurrentUser: false
      });

      subject.find(target).simulate('click');
      expect(props.openModal).toHaveBeenCalledTimes(1);
    });
  });

  // TODO: this test is breaking because of Tooltip library
  // Snapshot
  // it('renders correctly', () => {
  //   const subject = mountComponent({ ...props });
  //   const tree = renderer.create(subject).toJSON();
  //   expect(tree).toMatchSnapshot();
  // });
});
