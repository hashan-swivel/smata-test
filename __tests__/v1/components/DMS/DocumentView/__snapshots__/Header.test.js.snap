// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvoicePreview > Header renders correctly 1`] = `
<div
  className="document-header"
>
  <div
    className="invoice-header-container"
  >
    <div
      className="breadcrumbs"
    >
      <a
        className="breadcrumb"
        href="/documents?showInvoice=true"
        onClick={[Function]}
        onMouseEnter={[Function]}
        style={
          Object {
            "backgroundImage": "",
          }
        }
        title={null}
      >
        <span>
          Invoices
        </span>
      </a>
      <span
        className="breadcrumb"
      >
        10  Invoice   Contractor
      </span>
    </div>
    <div
      className="invoice-approve"
    >
      <button
        className="approve-invoice-button button primary "
        onClick={[Function]}
        type="button"
      >
        Approve
      </button>
      <span
        className="invoice-options-dropdown null"
        onClick={[Function]}
        role="presentation"
      >
        ...
      </span>
    </div>
  </div>
  <div
    className="invoice-status approved"
  >
    approved for payment
  </div>
  <div
    className="document-header-row-top"
  >
    <div
      className="file-type-icon file-red medium"
    >
      pdf
    </div>
    <div
      className="li-content"
    >
      <h3>
        10  Invoice   Contractor
      </h3>
    </div>
    <div
      className="document-header-buttons"
    >
      <button
        className="button secondary"
        onClick={[MockFunction]}
        type="button"
      >
        Edit
      </button>
    </div>
  </div>
  <div
    className="document-view-block-content"
  >
    <div
      className="document-header-row"
    >
      <div
        className="li-before-content"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-building fa-w-14 fa-lg "
          data-icon="building"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M436 480h-20V24c0-13.255-10.745-24-24-24H56C42.745 0 32 10.745 32 24v456H12c-6.627 0-12 5.373-12 12v20h448v-20c0-6.627-5.373-12-12-12zM128 76c0-6.627 5.373-12 12-12h40c6.627 0 12 5.373 12 12v40c0 6.627-5.373 12-12 12h-40c-6.627 0-12-5.373-12-12V76zm0 96c0-6.627 5.373-12 12-12h40c6.627 0 12 5.373 12 12v40c0 6.627-5.373 12-12 12h-40c-6.627 0-12-5.373-12-12v-40zm52 148h-40c-6.627 0-12-5.373-12-12v-40c0-6.627 5.373-12 12-12h40c6.627 0 12 5.373 12 12v40c0 6.627-5.373 12-12 12zm76 160h-64v-84c0-6.627 5.373-12 12-12h40c6.627 0 12 5.373 12 12v84zm64-172c0 6.627-5.373 12-12 12h-40c-6.627 0-12-5.373-12-12v-40c0-6.627 5.373-12 12-12h40c6.627 0 12 5.373 12 12v40zm0-96c0 6.627-5.373 12-12 12h-40c-6.627 0-12-5.373-12-12v-40c0-6.627 5.373-12 12-12h40c6.627 0 12 5.373 12 12v40zm0-96c0 6.627-5.373 12-12 12h-40c-6.627 0-12-5.373-12-12V76c0-6.627 5.373-12 12-12h40c6.627 0 12 5.373 12 12v40z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      <div
        className="li-content"
      >
        <a
          href="/building-profile?id=SP7110"
          style={
            Object {
              "color": "#4A90E2",
            }
          }
        >
          SP285671 - 2 Cross Street, MOSMAN, 2088, NSW
        </a>
      </div>
    </div>
    <div
      className="document-header-row"
    >
      <div
        className="li-before-content"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-dollar-sign fa-w-9 fa-lg "
          data-icon="dollar-sign"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 288 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M209.2 233.4l-108-31.6C88.7 198.2 80 186.5 80 173.5c0-16.3 13.2-29.5 29.5-29.5h66.3c12.2 0 24.2 3.7 34.2 10.5 6.1 4.1 14.3 3.1 19.5-2l34.8-34c7.1-6.9 6.1-18.4-1.8-24.5C238 74.8 207.4 64.1 176 64V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48h-2.5C45.8 64-5.4 118.7.5 183.6c4.2 46.1 39.4 83.6 83.8 96.6l102.5 30c12.5 3.7 21.2 15.3 21.2 28.3 0 16.3-13.2 29.5-29.5 29.5h-66.3C100 368 88 364.3 78 357.5c-6.1-4.1-14.3-3.1-19.5 2l-34.8 34c-7.1 6.9-6.1 18.4 1.8 24.5 24.5 19.2 55.1 29.9 86.5 30v48c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16v-48.2c46.6-.9 90.3-28.6 105.7-72.7 21.5-61.6-14.6-124.8-72.5-141.7z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      <div
        className="li-content"
      >
        <strong>
          Admin Fund: 
        </strong>
        <span>
          $
          -61,835.37
        </span>
          |  
        <strong>
          Capital Works Fund: 
        </strong>
        <span>
          $
          11,510.29
        </span>
          |  
        <strong>
          Balance: 
        </strong>
        <span
          style={
            Object {
              "color": "red",
            }
          }
        >
          $
          -50,325.08
        </span>
          
      </div>
    </div>
    <div
      className="document-header-row"
    >
      <div
        className="li-before-content"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-user fa-w-14 fa-lg "
          data-icon="user"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      <div
        className="li-content"
      >
        George Vumbaca (undefined - undefined)
      </div>
    </div>
  </div>
</div>
`;
